src/client/encode.rs

```diff
@@ -41,6 +41,14 @@ impl Encoder {
             url.push_str(query);
         }
 
+        if req.method() == Method::Connect {
```

Comment:
Oh nice one, I didn't know about this. It'd be helpful if we could add comment *why* we're doing this, linking to the RFC:

```suggestion
        // A client sending a CONNECT request MUST consists of only the host
        // name and port number of the tunnel destination, separated by a colon.
        // See: https://tools.ietf.org/html/rfc7231#section-4.3.6
        if req.method() == Method::Connect {
```